test_pull_requests

```diff
@@ -1731,11 +1982,11 @@ popd
           comments = get_comments(req['number'], req['base']['repo']['name'])
 
           bot_comment = get_comment_with_prefix(req['number'], req['base']['repo']['name'], settings['test_prefix'], comments)
-          if bot_comment && (bot_comment['body'] =~ /^#{settings['test_prefix']} Waiting/)
+          if bot_comment && (queue_waiting_state?(extract_bot_comment_fields(bot_comment['body'], settings)[:state]))
```

Comment:
Really? This is the only place this result of `extract_bot_comment_fields` is going to be used.